src/pool/srv_target.c

```diff
@@ -585,12 +585,21 @@ pool_hdl_key_cmp(struct d_hash_table *htable, d_list_t *rlink,
 	return uuid_compare(hdl->sph_uuid, key) == 0;
 }
 
+static uint32_t
+pool_hdl_key_hash(struct d_hash_table *htable, const void *key,
+		  unsigned int ksize)
+{
+	D_ASSERTF(ksize == sizeof(uuid_t), "%u\n", ksize);
+	return *((const uint32_t *)key);
+}
+
 static uint32_t
 pool_hdl_rec_hash(struct d_hash_table *htable, d_list_t *link)
 {
 	struct ds_pool_hdl *hdl = pool_hdl_obj(link);
+	uint32_t *retp = (uint32_t *)hdl->sph_uuid;
 
-	return d_hash_string_u32((const char *)hdl->sph_uuid, sizeof(uuid_t));
+	return *retp;
```

Comment:
whole UUID (128 bits) is suppose to be random enough and unique, but not sure if the first 32 bits taken here is random enough. Seems it depends on UUID algorithm.

Reply:
We have to limit a final value to 32-bit anyway. Then it will be shrinked to hash table size. So, I think we anyway loose randomness. The goal of this patch is to reduce performance impact of obtaining the hash value.